Front Door publicpreview

jmelvinwork opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

jmelvinwork committed:
Front Door public preview

Front Door public preview

jmelvinwork committed:
Merge pull request #1 from jmelvinwork/frontdoor-public-preview

Front Door public preview

daschult commented on specification/frontdoor/resource-manager/readme.nodejs.md:
```diff
@@ -0,0 +1,15 @@
+nodejs:
+  azure-arm: true
+  package-name: azure-arm-frontdoor
+  package-version: 5.2.0
```
Please remove this package-version property. We're now handling this in https://www.github.com/azure/azure-sdk-for-node.

daschult replied:
Nevermind. I'll take care of this file after this PR is merged.

daschult commented on specification/frontdoor/resource-manager/readme.nodejs.md:
```diff
@@ -0,0 +1,15 @@
+  package-version: 5.2.0
+  generate-license-txt: true
+  generate-package-json: true
+  generate-readme-md: false
```
This looks like a new service. In that case, I recommend replacing these three properties (`generate-license-txt`, `generate-package-json`, and `generate-readme-md`) to just `generate-metadata: true`.

daschult commented on specification/frontdoor/resource-manager/readme.nodejs.md:
```diff
@@ -0,0 +1,15 @@
+  generate-license-txt: true
+  generate-package-json: true
+  generate-readme-md: false
+  output-folder: $(node-sdks-folder)/lib/services/networkManagement2/lib
```
Please change the `output-folder` property to `$(node-sdks-folder)/lib/services/frontdoorManagement`.

azuresdkci commented:
Can one of the admins verify this patch?

jmelvinwork commented:
This was signed off in the internal repo already.

KrisBash reviewed (approved):
Sign off from ARM

jmelvinwork commented:
@lmazuel Can you please review this today? We want to get this merged ASAP

jmelvinwork commented:
The linter says the following:
_Potential new ARM errors
========================
R3023 - OperationsAPIImplementation
Operations API must be implemented for '/providers/Microsoft.Network/operations'.
  at specification/frontdoor/resource-manager/Microsoft.Network/preview/2018-08-01-preview/webapplicationfirewall.json:36_

But we are using NRP's Microsoft.Network/operations API, which is defined under /network/...
How do we fix this error?

jmelvinwork committed:
Add suppression for operationsAPIImplementation as we use NRP's operations/ API

Add suppression for operationsAPIImplementation as we use NRP's operations/ API

jmelvinwork committed:
Merge branch 'master' of https://github.com/jmelvinwork/azure-rest-api-specs into frontdoor-public-preview

jmelvinwork commented:
> # The linter says the following:
> _Potential new ARM errors
> R3023 - OperationsAPIImplementation
> Operations API must be implemented for '/providers/Microsoft.Network/operations'.
> at specification/frontdoor/resource-manager/Microsoft.Network/preview/2018-08-01-preview/webapplicationfirewall.json:36_
> 
> But we are using NRP's Microsoft.Network/operations API, which is defined under /network/...
> How do we fix this error?

Laurent helped me out. Thanks, Laurent!

jmelvinwork commented:
Did not mean to close PR! Sorry

jmelvinwork committed:
Fix typo

Fix typo

jmelvinwork committed:
Fix indentation

Fix indentation

lmazuel commented on vim.exe.stackdump:
```diff
@@ -0,0 +1,19 @@
+Stack trace:
```
Please remove that file.

lmazuel commented on specification/frontdoor/resource-manager/Microsoft.Network/preview/2018-08-01-preview/webapplicationfirewall.json:
```diff
@@ -0,0 +1,619 @@
+        }
+      ]
+    },
+    "WebApplicationFirewallPolicyPropertiesFormat": {
```
Three of your examples defines `resourceState` as should be part of this model, but it's not.

Being that the right thing to do is to model PRECISELY the actual rest API:
- Add resourceState here is this really exists
- Remove it from the examples if this does not really exists (has been removed from API, etc.)

Impacted examples:
- specification/frontdoor/resource-manager/Microsoft.Network/preview/2018-08-01-preview/examples/WafPolicyCreateOrUpdate.json
-  specification/frontdoor/resource-manager/Microsoft.Network/preview/2018-08-01-preview/examples/WafPolicyGet.json
- specification/frontdoor/resource-manager/Microsoft.Network/preview/2018-08-01-preview/examples/WafListPolicies.json

jmelvinwork committed:
Fix resourceState in definition

WebApplicationFirewallPolicy Swagger did not contain the return value ResourceState.

lmazuel committed:
Fix resourceState case

lmazuel committed:
Delete vim.exe.stackdump

lmazuel commented:
@jmelvinwork My last fixes fixed the CLI, and I think it's what you intended. Merging the PR since you told me you want it merge today.
Still, you own this, please double-check my commits and let me know if you have any concerns. Ideally, let me know if you don't have concerns too :)
Thanks!
